meteor/sofie-server/data-access/repositories/test/mongo-test-database.ts

```diff
@@ -0,0 +1,91 @@
+// eslint-disable-next-line node/no-unpublished-import
+import { MongoMemoryServer } from 'mongodb-memory-server'
+import { Db, MongoClient } from 'mongodb'
+import { Rundown } from '../../../model/entities/rundown'
+import { MongoEntityConverter, MongoRundown } from '../mongo/mongo-entity-converter'
+import { Segment } from '../../../model/entities/segment'
+import { Part } from '../../../model/entities/part'
+import { Piece } from '../../../model/entities/piece'
+import { MongoDatabase } from '../mongo/mongo-database'
+import { when } from 'ts-mockito'
+
+export class MongoTestDatabase {
+	private mongoServer: MongoMemoryServer
+	private client: MongoClient
+
+	constructor() {
+		// Set a timeout beyond the default of 5 Seconds to ensure CI tests don't exceed the limit on GitHub
+		jest.setTimeout(15000)
```

Comment:
I presume creating an in-memory database it the lengthy operation here. Instead of spinning up a new one for every .spec. file, can one database somehow be reused?

Reply:
@RasmusAlbrektsen can propably explain it better. 
He was the one to originally add that for the tests that runs in the in memory database.

From my understanding however, it is becuase the CI was timing out while running these tests.
You might be right about spinning up the DB, taking some time.

Only creating the database ones might help, but risks tests interfering with one another, which is why it is heavily discouraged to do so. Even if done so, that wouldn't allow us to remove the commented on line, as far as i know, so it wouldn't have any benefits in that regard.

Reply:
It is definately possible.

One thing we just need to be aware of is, that each test don't run on data that has been populated/modified by another test, because this leads to potentially random failing test.

One way to avoid it is by creating a new database instance for each test, which has been done here. Since that is a heavy operation it might not be the most suitable approach.

Another way would be to reuse the same database instance for each test, but to completely wipe it for data between test and then repopulate it. This could however create potential problems if the tests are running at the same since one test might populate the database, then another test wipes it in its cleanup, then the first test tries to read data from the database that now has been wiped. But this would be avoided by not running the test asynchronized.

What is most important here is to ensure that we have a "fresh" starting point for each test that other tests can't modify. If you have suggestions for that we are all ears :)